src/egraph.rs

```diff
@@ -406,9 +412,21 @@ impl<L: Language, M: Metadata<L>> EGraph<L, M> {
     }
 
     fn rebuild_classes(&mut self) -> usize {
+	let (find, mut_values) = self.classes.split();
+
+	let mut sum: usize = 0;
+	for ids in self.sigtoclasses.values_mut() {
+	    sum += ids.len();
+	    let unique: IndexSet<Id> = ids
+		.iter()
+		.map(|id| find(*id))
+		.collect();
+	    ids.clear();
+	    ids.extend(unique);
+	}
+	
```

Comment:
I think this is why it's failing tests. This doesn't capture enodes added during a metadata `modify`. This should probably just scan the eclasses (perhaps simultaneously with the loop below), rebuilding the map entirely.

Reply:
Oh yes, I forgot to fix that. Thanks

Reply:
Rebuilding the map entirely would mean that you can only search directly after rebuilding. Right now rebuilding is optional before you search.
Would making an add_enode function for modify work better?

Reply:
(kind of optional, you don't get the congruence guarantee)

Reply:
Ah but then modify couldn't prune the nodes. I'll change it then.

Reply:
rebuilding is not optional before you search, so you can assume it's been called